newrelic/terraform-provider-newrelic
# Feat/nrql condition threshold operators

# Description

Adds support for 3 additional term threshold operators for static NRQL conditions.

See also: https://github.com/newrelic/newrelic-client-go/pull/886

## Type of change


## Checklist:

Please delete options that are not relevant.


## How to test this change?

With the updated New Relic Go client and these changes, we should be able to run a `terraform apply` with the following NRQL condition resource:

```hcl
variable "nr_api_key" {
  type        = string
  description = "your new relic API key"
}

variable "account_id" {
  type        = number
  description = "your new relic account"
}

variable "region" {
  type        = string
  description = "the region you're testing in"
}

provider newrelic {
  api_key    = var.nr_api_key
  account_id = var.account_id
  region     = var.region
}

resource "newrelic_alert_policy" "some_policy" {
  name                = "Terraform test"
  incident_preference = "PER_POLICY"
}

resource "newrelic_nrql_alert_condition" "some_condition" {
  name      = "SomeTF Condition"
  policy_id = newrelic_alert_policy.some_policy.id
  nrql {
    query = "SELECT max(duration) FROM Transaction"
  }
  critical {
    operator              = "above_or_equals"
    threshold             = 0
    threshold_duration    = 120
    threshold_occurrences = "all"
  }
  aggregation_window             = 60
  close_violations_on_expiration = true
  violation_time_limit_seconds   = 3600
  aggregation_method             = "event_flow"
  aggregation_delay              = 120
  type                           = "static"
  expiration_duration            = 180
  enabled                        = false
}
```

The 3 `critical.operator` values being added are:

- `above_or_equals`
- `below_or_equals`
- `not_equals`
